python/test/unit_distributed/runner.py

```diff
@@ -21,16 +21,22 @@
     import options
     import test_contexts_arbmpi
     import test_contexts_mpi4py
+    import test_domain_decompositions_mpi4py
```

Comment:
Can you refactor the tests so that you can test either arbmpi or mpi4py with the same test?
Or maybe only test all the features using arbmpi, and have separate test that verifies that mpi4py can be used to initialize an Arbor context (but don't then verify all the features working with mpi4py)?

Reply:
I think I can refactor.

Reply:
I would refactor `test_domain_decomposition` to test either arbmpi or mpi4py (dependend on which is used/set up), but I would keep the context tests seperated to force arbmpi/mpi4py being tested individually. Is that fine?

Reply:
Yes. The context tests should be separate, to ensure that we can use both.

Can we just run all the rest of the MPI tests with `arbmpi`?

Reply:
Do you force the rest of the MPI to be tested with `arbmpi`?

Reply:
Done.